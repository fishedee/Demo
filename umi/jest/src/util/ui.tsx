import React from 'react';

type ButtonProps = {
    title:string;
    onClick?:()=>void;
}
const Button:React.FC<ButtonProps> = (props)=>{

    return (<button onClick={props.onClick}>{props.title}</button>);
}

type ButtonList = {
    title:string;
    list:ButtonProps[]
}

const ButtonList:React.FC<ButtonList> = (props)=>{
    return (<div>
        <h2>{props.title}</h2>
        {props.list.map((single,index)=>{
            return (<Button key={index} {...single}/>);
        })}
    </div>);
}

export {
    Button,
    ButtonList
};