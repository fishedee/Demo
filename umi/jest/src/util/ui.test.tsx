import React from 'react'
import Enzyme, { shallow ,render} from 'enzyme'
import Adapter from 'enzyme-adapter-react-16'
import toJson from 'enzyme-to-json'
import {Button,ButtonList} from './ui';

Enzyme.configure({ adapter: new Adapter() })


test('test1',()=>{
    //浅渲染
    const MyButton = shallow(<Button title={'按钮啊'}/>);

    //拿出text
    const buttonText = MyButton.text();

    expect(buttonText).toEqual('按钮啊');
})


test('test2',()=>{

    const myClick = jest.fn();
    //浅渲染
    const MyButton = shallow(<Button title={'按钮啊'} onClick={myClick}/>);

    //模拟事件点击
    MyButton.simulate('click');

    // 期望调用了这个方法
    expect(myClick).toHaveBeenLastCalledWith();
})

test('test3',()=>{
      //浅渲染
      const app = shallow(<ButtonList title="12" list={[
          {
              title:'button1',
          },
          {
              title:'button2',
          }
      ]}/>);

      //拿出text
      const buttonLength = app.find('button').length;
  
      expect(buttonLength).toEqual(0);
})

test('test4',()=>{
    //深渲染，只渲染为html
    const app = render(<ButtonList title="12" list={[
        {
            title:'button1',
        },
        {
            title:'button2',
        }
    ]}/>);

    //拿出text
    const buttonLength = app.find('button').length;

    expect(buttonLength).toEqual(2);
})

test('test5',()=>{
    //深渲染，只渲染为html
    const app = render(<ButtonList title="12" list={[
        {
            title:'button1',
        },
        {
            title:'button2',
        }
    ]}/>);

    expect(toJson(app)).toMatchSnapshot();
})