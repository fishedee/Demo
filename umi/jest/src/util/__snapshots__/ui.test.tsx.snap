// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test5 1`] = `
<div>
  <h2>
    12
  </h2>
  <button>
    button1
  </button>
  <button>
    button2
  </button>
</div>
`;
